diff --git a/lib/module/initializers.js b/lib/module/initializers.js
index d012b5093956744fac0aa3d7fe9df734b20d191a..b921a34860fc6c86f1e1578ec61bc1078745201c 100644
--- a/lib/module/initializers.js
+++ b/lib/module/initializers.js
@@ -119,7 +119,7 @@ function initializeRNRuntime() {
 function initializeWorkletRuntime() {
   if (globalThis._WORKLETS_BUNDLE_MODE) {
     setupCallGuard();
-    if (__DEV__) {
+    if (__DEV__ && !SHOULD_BE_USE_WEB) {
       /*
        * Temporary workaround for Metro bundler. We must implement a dummy
        * Refresh module to prevent Metro from throwing irrelevant errors.
diff --git a/src/initializers.ts b/src/initializers.ts
index dc252b387093cf6839a97f0d08e96581839dab16..fd180700c777e02ffa056a497ad6b10f43007de8 100644
--- a/src/initializers.ts
+++ b/src/initializers.ts
@@ -117,7 +117,7 @@ function initializeRuntime() {
 
 /** A function that should be run only on React Native runtime. */
 function initializeRNRuntime() {
-  if (__DEV__) {
+  if (__DEV__ && !SHOULD_BE_USE_WEB) {
     const testWorklet = () => {
       'worklet';
     };
